src/containers/Assistants/CreateAssistant/CreateAssistant.tsx

```diff
@@ -0,0 +1,260 @@
+
+import CopyIcon from 'assets/images/CopyGreen.svg?react';
+
+import { AssistantOptions } from '../AssistantOptions/AssistantOptions';
+
+import styles from './CreateAssistant.module.css';
+
+interface CreateAssistantProps {
+  currentId: string | number | null;
+  updateList: boolean;
+  setCurrentId: any;
+  setUpdateList: any;
+}
+
+export const CreateAssistant = ({
+  currentId,
+  setUpdateList,
+  setCurrentId,
+  updateList,
+}: CreateAssistantProps) => {
+  const [assistantId, setAssistantId] = useState('');
+  const [name, setName] = useState('');
+  const [model, setModel] = useState<any>(null);
+  const [instructions, setInstructions] = useState('');
+  const [options, setOptions] = useState({ fileSearch: true, temperature: 1 });
+  const [showConfirmation, setShowConfirmation] = useState(false);
+
+  const states = {
+    name,
+    model,
+    instructions,
+    options,
+  };
+
+  let modelOptions = [];
+
+  const { data: modelsList, loading: listLoading } = useQuery(GET_MODELS);
+  const [getAssistant, { loading, data }] = useLazyQuery(GET_ASSISTANT);
+
+  const [updateAssistant, { loading: savingChanges }] = useMutation(UPDATE_ASSISTANT);
+
+  const [deleteAssistant, { loading: deletingAssistant }] = useMutation(DELETE_ASSISTANT);
+
+  if (modelsList) {
+    modelOptions = modelsList?.listOpenaiModels.map((item: string, index: number) => ({
+      id: index.toString(),
+      label: item,
+    }));
+  }
+
+  useEffect(() => {
+    if (currentId && modelsList) {
+      getAssistant({
+        variables: { assistantId: currentId },
+        onCompleted: ({ assistant }) => {
+          setAssistantId(assistant?.assistant?.assistantId);
+          setName(assistant?.assistant?.name);
+          let modelValue = modelOptions?.find(
+            (item: any) => item.label === assistant?.assistant?.model
+          );
+          setModel(modelValue);
+          setInstructions(assistant?.assistant?.instructions || '');
+          setOptions({
+            ...options,
+            temperature: assistant?.assistant?.temperature,
+          });
+        },
+      });
+    }
+  }, [currentId, modelsList]);
+
+  const handleCreate = () => {
+    const {
+      instructions: instructionsValue,
+      model: modelValue,
+      name: nameValue,
+      options: optionsValue,
+    } = states;
+
+    const payload = {
+      instructions: instructionsValue,
+      model: modelValue.label,
+      name: nameValue,
+      temperature: optionsValue?.temperature,
+    };
+
+    updateAssistant({
+      variables: {
+        updateAssistantId: currentId,
+        input: payload,
+      },
+      onCompleted: () => {
+        setNotification('Changes saved successfully', 'success');
+        setUpdateList(!updateList);
+      },
+      onError: (errors) => {
+        setErrorMessage(errors);
+      },
+    });
+  };
+
+  const formFields: any = [
+    {
+      component: AutoComplete,
+      name: 'model',
+      options: modelOptions || [],
```

Comment:
it looks like changing value of this autocomplete leads to lot of warnings, please fix.